docs/default.nix

```diff
@@ -0,0 +1,57 @@
+{ pkgs ? import ../pkgs { } }:
+
+with pkgs.lib;
+let
+  eval = import ../default.nix { inherit pkgs; configuration = { }; };
+
+  robotnixOptionsDoc = pkgs.nixosOptionsDoc {
+    inherit (eval) options;
+  };
+
+  optionsMd =
+    let
+      options = robotnixOptionsDoc.optionsNix;
+    in
+    concatStrings (map
+      (name:
+        let
+          option = options.${name};
+          body = ''
+            ${option.description}
+
+          '' + optionalString (option ? default) ''
+            Default: `${builtins.toJSON option.default}`
+
+          '' + optionalString (option ? example) ''
+            Example: `${builtins.toJSON option.example}`
```

Comment:
There is one gotcha here, namely that `builtins.toJSON` will not generate valid Nix code for non-empty lists and non-empty attrsets.  The situation is probably also problematic for strings with embedded control characters.  However, `nixosOptionsDoc.optionsJSON` has the same problem.  Does nixpkgs contain a function for serializing data as valid Nix?  At the same time, as long as none of the corner cases named above apply, using `builtins.toJSON` should be just fine.

Reply:
I found `lib.generator.toPretty` which comes very close:
```nix
nix-repl> v = { a = [ 1 2 3.14159265 { b = ''x\ny''; } ]; c = "z"; d = x: x + 1; }

nix-repl> lib.generators.toPretty {} v                                             
"{ \"a\" = [ 1 2 ~3.141593 { \"b\" = \"x\\ny\"; } ]; \"c\" = \"z\"; \"d\" = <λ>; }"
```
It's not so nice that keys in the attrset are always quoted, but at least it's valid Nix.  Functions of course cannot be reasonably serialized so representing them by `<λ>` is quite okay.  The only real problem here is the serialization of floating point numbers.  I don't see why it wouldn't just print the internal representation.